[spmd] move the param_buffers to the front

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #98452
* #98866
* __->__ #98437

Makes it a bit easy to track the parameter count and corresponding named
states

wanchaol committed:
[spmd] move the param_buffers to the front

Makes it a bit easy to track the parameter count and corresponding named
states

[ghstack-poisoned]

mrshenli reviewed (commented):
Thanks @wanchaol, I think this change is already landed this morning: https://github.com/pytorch/pytorch/pull/98392

wanchaol commented:
> Thanks @wanchaol, I think this change is already landed this morning: #98392

Oh yes, this PR is actually on top of @yifuwang's change, and moving the parameters to the first position instead of named_states to the first position, I am not sure what order PT2 gonna decide when including optimizer to the graph, but from my prospective it's easier to first manage parameter then manage the optim states :) Happy to discuss further on this

wanchaol committed:
Update on "[spmd] move the param_buffers to the front"


Makes it a bit easy to track the parameter count and corresponding named
states

[ghstack-poisoned]

wanchaol committed:
Update on "[spmd] move the param_buffers to the front"


Makes it a bit easy to track the parameter count and corresponding named
states

[ghstack-poisoned]
